### PR TITLE
[Merged by Bors] - Fix a bug in fork pruning

### PR DESCRIPTION
**Pull request by adaszko**
Extracted from https://github.com/sigp/lighthouse/pull/1380 because merging #1380 proves to be contentious.

### CODE REVIEW

File: beacon_node/beacon_chain/src/migrate.rs [hunk 1]
```diff
@@ -216,6 +211,11 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> Migrate<E, Hot, Cold>
         ) {
             eprintln!("Pruning error: {:?}", e);
         }
+
+        if let Err(e) = process_finalization(self.db.clone(), state_root, &new_finalized_state) {
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/src/migrate.rs:215]: The same change in ordering needs to happen in the `BackgroundMigrator` as well

File: beacon_node/beacon_chain/tests/store_tests.rs [hunk 2]
```diff
@@ -743,7 +743,7 @@ fn prunes_abandoned_fork_between_two_finalized_checkpoints() {
     let (stray_blocks, stray_states, _, stray_head, _) = harness.add_stray_blocks(
         harness.get_head_state(),
         slot,
-        slots_per_epoch - 1,
+        slots_per_epoch - 3,
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/tests/store_tests.rs:746]: I can't quite work out why this needs to be `- 3`, can you remind me?

Comment (adaszko) [reply beacon_node/beacon_chain/tests/store_tests.rs:746]: It's in the commit messages.

File: beacon_node/beacon_chain/src/migrate.rs (hunk 1 above)
Comment (adaszko) [reply beacon_node/beacon_chain/src/migrate.rs:215]: That's a very good point

**Comment by adaszko** [comment]
r+

**Pull request closed without merging**